File: ios/MeiGei/MeiGei/Models/CustomFood.swift
import Foundation
import SwiftData

/// 自定义食材（参与同步，对应后端 custom_food）。
/// 标准库（~1500 条）随包 seed、只读，不在此模型内。
@Model
final class CustomFood: Syncable {
    @Attribute(.unique) var localId: UUID
    var serverId: UUID?
    var updatedAt: Date
    var deletedAt: Date?
    var version: Int
    var syncStatusRaw: String

    var name: String
    /// 来源标识：authority | personal。个人项不进标准库首屏（design.md D9）。
    var source: String
    /// 营养含量基准，如 "100g" / "1 份"。
    var unitBasis: String
    var kcal: Double?
    var proteinG: Double?
    var carbG: Double?
    var fatG: Double?
    var fiberG: Double?
    var sugarG: Double?
    var sodiumMg: Double?

    init(
        localId: UUID = UUID(),
        name: String,
        source: String = "personal",
        unitBasis: String = "100g",
        kcal: Double? = nil,
        proteinG: Double? = nil,
        carbG: Double? = nil,
        fatG: Double? = nil,
        fiberG: Double? = nil,
        sugarG: Double? = nil,
        sodiumMg: Double? = nil,
        now: Date = .now
    ) {
        self.localId = localId
        self.serverId = nil
        self.updatedAt = now
        self.deletedAt = nil
        self.version = 0
        self.syncStatusRaw = SyncStatus.pendingCreate.rawValue
        self.name = name
        self.source = source
        self.unitBasis = unitBasis
        self.kcal = kcal
        self.proteinG = proteinG
        self.carbG = carbG
        self.fatG = fatG
        self.fiberG = fiberG
        self.sugarG = sugarG
        self.sodiumMg = sodiumMg
    }
}

File: backend/src/main/java/com/meigei/nutrition/entity/CustomFood.java
package com.meigei.nutrition.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.meigei.common.entity.BaseEntity;
import com.meigei.sync.UserOwned;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * 用户自定义食材（搜不到权威库时补充）。source 仅 'personal'，
 * 权威库为内置 seed 目录不入库。7 项营养素按 unitBasis（默认 100g）计。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("custom_food")
public class CustomFood extends BaseEntity implements UserOwned {

    private UUID userId;

    private String name;

    private String source;

    private String unitBasis;

    private BigDecimal kcal;

    private BigDecimal proteinG;

    private BigDecimal carbG;

    private BigDecimal fatG;

    private BigDecimal fiberG;

    private BigDecimal sugarG;

    private BigDecimal sodiumMg;
}
